force-app/main/default/lwc/donationHistoryTable/donationHistoryTable.js

```diff
@@ -1,81 +1,90 @@
+        this.totalNumberOfRows = data.donations.length;
+        this.paymentMethodLabel = data.paymentMethodLabel;
+        this.arePaymentsEnabled = data.isPaymentsEnabled;
 
-    @wire(getObjectInfo, { objectApiName: DATA_IMPORT })
-    oppInfo({ data, error }) {
-        if (data) {
-            this.paymentMethodLabel = data.fields[PAYMENT_FIELD.fieldApiName].label
-        };
         this.columns = [
-            { label: RD2_ScheduleVisualizerColumnDate, fieldName: 'closeDate', type: 'date-local', typeAttributes:{
+            {
+              label: RD2_ScheduleVisualizerColumnDate,
+              fieldName: "closeDate",
+              type: "date-local",
+              typeAttributes: {
                 year: "numeric",
                 month: "numeric",
                 day: "numeric",
+              },
+              cellAttributes: { alignment: "right" },
+            },
+            {
+              label: donationHistoryDonorLabel,
+              fieldName: "name",
+              type: "text",
             },
-            cellAttributes: { alignment: 'right' }},
-            { label: donationHistoryDonorLabel, fieldName: 'name', type: 'text' },
-            { label: commonAmount, fieldName: 'amount', type: 'currency', },
-            { label: this.paymentMethodLabel, fieldName: 'paymentMethod', type: 'text', },
+            { label: commonAmount, fieldName: "amount", type: "currency" },
         ];
-    }
-    
-    // eslint-disable-next-line @lwc/lwc/no-async-await
-    async connectedCallback() {
-        getDonationHistory({contactId: this.contactId})
-        .then(data => {
-            if (data) {
-                this.allData = data;
-                this.data = data.slice(0, RECORDS_TO_LOAD);
-                this.totalNumberOfRows = data.length;
-            }
-        });
-    }
-
-    /**
-     * 
-     * @param {*} event 
-     * handle the scroll event to get more content and concat to the existing table data
-     */
-    loadMoreDonationData(event){
-        event.target.isLoading = true;
-        if (this.data.length >= this.totalNumberOfRows) {
-            event.target.enableInfiniteLoading = false;
-            event.target.isLoading = false;
-            return;
+        if (this.arePaymentsEnabled) {
+            this.columns.push(
+                { label: this.paymentMethodLabel, fieldName: "paymentMethod", type: "text", },
+            );
         }
-        const current = this.data.length;
-        const offset = current + RECORDS_TO_LOAD;
-        const currentData = this.data;
-        //Appends new data to the end of the table
-        const newData = currentData.concat(this.allData.slice(current, offset));
-        this.data = newData;
-        event.target.isLoading = false;
+      }
+    });
+  }
+
+  /**
+   *
+   * @param {*} event
+   * handle the scroll event to get more content and concat to the existing table data
+   */
+  loadMoreDonationData(event) {
+    event.target.isLoading = true;
```

Comment:
I don't think we need to specify `isLoading = true` in this situation, this block of code should run instant/synchronous, there aren't any callouts to apex methods. In the future, are we going to have a query that runs in here to query additional data?

Reply:
Hi Benjamin, I do not think that's the case. However, this particular block of code was untouched by myself. There is **[a ticket already being worked](https://github.com/SalesforceFoundation/NPSP/pull/6814/)** on related to Infinite Scrolling which modifies this logic. Maybe this modification should be done there?